### PR TITLE
[infra] Use kotlin standard libraries from the toolchain, as opposed to hard coded.

### PR DESCRIPTION
**Pull request by restingbull**
Modifies the toolchain to provide a common set of stdlib for the builder and jdeps merger via the classpath advice on a java binary.

Also extends the `examples/trivial` to validate the built binary executes as part of testing the java/kotlin integration.

Aside from improving correctness, the hope is that this will fix the link error seen when upgrading to newer versions of kotlinc.

### CODE REVIEW

File: kotlin/internal/toolchains.bzl [hunk 1]
```diff
@@ -200,6 +193,10 @@ _kt_toolchain = rule(
                 "21",
             ],
         ),
+        "js_target": attr.string(
```
Comment (Bencodes) [inline kotlin/internal/toolchains.bzl:196]: This should have been removed

File: examples/trivial/BUILD [hunk 2]
```diff
@@ -27,3 +27,16 @@ ktlint_config(
     experimental_rules_enabled = False,
     visibility = ["//visibility:public"],
 )
+
+sh_test(
```
Comment (Bencodes) [inline examples/trivial/BUILD:31]: There's a skylib macro for this that asserts if something builds.

Probably worth using just to minimize the surface area of rules Kotlin a bit.

Comment (restingbull) [reply examples/trivial/BUILD:31]: Needs to run to test it. Building is easy -- but the kotlin/jvm integration will only fail if it doesn't run.

File: kotlin/internal/toolchains.bzl [hunk 3]
```diff
@@ -122,7 +126,7 @@ _kt_toolchain = rule(
         ),
         "language_version": attr.string(
             doc = "this is the -language_version flag [see](https://kotlinlang.org/docs/reference/compatibility.html)",
-            default = "2.0",
+            default = "1.9",
```
Comment (nkoroste) [inline kotlin/internal/toolchains.bzl:129]: why are we going back to an older version?

**Review by nkoroste — APPROVED** [review]

File: kotlin/internal/toolchains.bzl (hunk 3 above)
Comment (restingbull) [reply kotlin/internal/toolchains.bzl:129]: I can't merge?

**Comment by agluszak** [comment]
> Aside from improving correctness, the hope is that this will fix the link error seen when upgrading to newer versions of kotlinc.

Yes, this seems to work!

**Pull request merged**